node/src/components/consensus/highway_core/state.rs

```diff
@@ -543,6 +543,19 @@ impl<C: Context> State<C> {
             Some(current)
         })
     }
+
+    /// Returns the median round exponent of all the validators that haven't been observed to be
+    /// malicious, as seen by the current panorama.
+    /// Returns none if there are no correct validators in the panorama.
+    pub(crate) fn median_round_exp(&self) -> Option<u8> {
+        let mut round_exponents: Vec<u8> = self
+            .panorama
+            .iter_correct(self)
+            .map(|vote| vote.round_exp)
+            .collect();
+        round_exponents.sort();
```

Comment:
Since most round exponents will usually agree, it would probably be much faster to create a `BTreeMap<u8, usize>` instead, and count how often each occurs.

Reply:
Is it worth it? Performance difference will be negligible since it happens once per era. I would vote for the readability here.

Reply:
Sure, I'm fine with that, too.

On the other hand it was @fizyk20 who made me use `u32` in `ValidatorIndex`, in case we want more than 65536 validators. :grin:

Reply:
Looks good, but if @goral09 prefers, I'm fine with reverting it, too.

Reply:
Cool, nifty.